### ISSUE TITLE
Incorrect keyboard height when the keyboard is visible and the device orientation changes

### ISSUE THREAD

**Issue by DaveRand** [EddyVerbruggen/cordova-plugin-native-keyboard#31]
Hi Eddy,

I am currently testing the **native keyboard plugin** using an **iPhone 5s** and **iOS 10.2.1**.

I am trying to figure out how to get the correct keyboard height when the keyboard is initially visible in portrait orientation and the device is changed to landscape orientation.

When the keyboard is visible and the orientation of the device is changed from portrait to landscape, the **onKeyboardWillShow** and **onKeyboardDidShow** callbacks are both called and each receives the current keyboard height. I need an accurate keyboard height so I can change the height of a scrollable list which appears above the keyboard. The list expands or contracts when the keyboard becomes visible or hidden.

The first time the orientation is changed from portrait to landscape, the **onKeyboardWillShow** and **onKeyboardDidShow** callbacks are called, but they both return the old portrait keyboard height instead of the new landscape keyboard height. The height I need in order to resize the list is the new landscape keyboard height.

When the orientation is changed back to portrait, both the **onKeyboardWillShow** and **onKeyboardDidShow** callbacks are called, but they return the landscape keyboard height instead of the new portrait keyboard height.

I am able to get the portrait keyboard height if the app is initially opened in portrait orientation, but I cannot get the correct landscape keyboard height when I need it. Is there any way to get the landscape keyboard height when the keyboard is visible and the device is initially changed to device orientation?

Thanks

**Comment by DaveRand** [comment]
Here is some additional information:

The following callbacks are made when the keyboard is visible and you change the device orientation:

onKeyboardWillHide
onKeyboardDidHide
onKeyboardWillShow - the height returned is incorrect.
onKeyboardDidShow - the height returned is incorrect.

The following callbacks are made when the orientation remains the same and you close the keyboard and open it again:

onKeyboardWillHide
onKeyboardDidHide
onKeyboardWillShow - the height returned is correct.
onKeyboardDidShow - the height returned is correct.

**Comment by EddyVerbruggen** [comment]
Hi Dave, I've updated the iOS version. Do you want to test from GitHub, or do I need to publish to npm?

**Comment by DaveRand** [comment]
Thanks. I can test it from Github.


At 01:56 AM 3/16/2017, you wrote:

**Comment by DaveRand** [comment]
Hi Eddy, I tested the iOS version from GitHub.

The **onKeyboardDidShow** callback is getting called 2 additional times now. These are the callbacks that are made when the keyboard is visible and you change the device orientation:

onKeyboardWillHide
onKeyboardDidHide
onKeyboardWillShow - the height returned is for the prior orientation.
onKeyboardDidShow - the height returned is for the prior orientation.
onKeyboardDidShow - the height returned is for the prior orientation.
onKeyboardDidShow - the height returned is for the current orientation.

I should be able to ignore the extra call to **onKeyboardDidShow** which returns the prior orientation a second time and just use the last call which returns the current orientation.

Thanks

**Comment by DaveRand** [comment]
Hi Eddy,

The **iOS version** of my app is now resizing correctly to accommodate the difference in keyboard height on an orientation change.

Thank you

**Comment by EddyVerbruggen** [comment]
Ah, that's great to hear!

And you're right, that last `onKeyboardDidShow` event is the one you need, the others are triggered natively as well and are a bit hard to suppress.

You have the same issue on Android as well, right? I tested with my and Android 4.4.4 tablet and the keyboard height was correct after orientation changes..

**Comment by DaveRand** [comment]
I will test it again using a couple of Android phones.

**Comment by DaveRand** [comment]
I just tested the new plugin using an **HTC One Mini (Android 4.4.2)** which is not a very fast device, but I've seen worse. The only event that fires when the keyboard is open and the orientation changes from portrait to landscape is **onKeyboardDidHide** which does not return a height parameter.

I tested the app using the plugin on NPM but I get the same results. Does NPM still have the prior version?

Thanks

**Comment by EddyVerbruggen** [comment]
Yes npm is on the previous version. But the only difference with GitHub is the iOS keyboard height fix. Nothing changed for Android. Perhaps a simulator can reproduce what you're seeing for me on Android..

**Comment by DaveRand** [comment]
I'm having someone test it for me on a faster 
Android. I will update the results shortly.

Does your device fire any other events besides 
onKeyboardDidHide when the keyboard is visible 
and you change the orientation to landscape?



At 11:43 AM 3/16/2017, you wrote:

**Comment by EddyVerbruggen** [comment]
On my Android 4.4.4 tablet I get `keyboardDidShow` after rotation, with the correct height.

**Comment by EddyVerbruggen** [comment]
1.3.2 is now on npm - with the same iOS fix as you've already tested. No change for Android (yet).

**Comment by DaveRand** [comment]
Hi Eddy,

I had someone test a different Android device for me and it appears to be working. The **onKeyboardDidHide** and **onKeyboardDidShow** events are both fired each time the device is rotated. The height returned in landscape orientation is a lesser value than the height returned in portrait orientation which is what should be expected.

For some reason, my **HTC One Mini** does not receive the **onKeyboardDidShow** event in landscape orientation, but does receive it in portrait orientation.

**Comment by EddyVerbruggen** [comment]
Android 🔫

**Comment by DaveRand** [comment]
Something interesting about the HTC One Mini keyboard in landscape orientation is that it fills the entire screen. Also, the left button goes away and the right button label somehow changes from "Send" to "Done". 


Here is the landscape version:

![landscape_keyboard](https://cloud.githubusercontent.com/assets/4587163/24041471/db1d1fe8-0aca-11e7-9943-f39e22a61957.png)



Here is the portrait version:

![portrait_keyboard2](https://cloud.githubusercontent.com/assets/4587163/24041480/e3e92a2c-0aca-11e7-92f1-fa984c6ae49e.png)
